src/utils/__test__/date.test.ts

```diff
@@ -0,0 +1,11 @@
+import { describe, it } from 'node:test';
+
+import assert from 'assert';
+
+import { getPacificAucklandYearMonthDay } from '../date.js';
+
+describe('getPacificAucklandYearMonthDay', () => {
+  it('should format as yyyy-mm-dd', () => {
+    assert.equal(getPacificAucklandYearMonthDay('2012-12-31T11:00:00Z'), '2013-01-01');
```

Comment:
We probably want at least two more tests, to make sure time zones are handled properly:

- 2012-12-31T10:59:59Z (just before Pacific/Auckland midnight)
- 2012-06-15T11:59:59Z (just before Pacific/Auckland midnight in winter time)
- 2012-06-15T12:00:00Z (just at Pacific/Auckland midnight in winter time)